kura/org.eclipse.kura.container.orchestration.provider/src/main/java/org/eclipse/kura/container/orchestration/provider/DockerService.java

```diff
@@ -22,34 +22,44 @@
  */
 public interface DockerService {
 
-    /**
-     * Lists all available containers, like running 'Docker ps -a'
-     *
-     * @return
-     */
-    public String[] listContainers();
-
     /**
      * Lists all available containers by id only, like running 'Docker ps -a'
      *
      * @return
      */
-    public String[] listContainersByID();
+    public List<String> listContainersIds();
```

Comment:
There are many Legacy API functions like this that were created before ContainerDescriptor was realized. I think that going forward we can remove all of these, and rely only on the ContainerDescror API methods.

Reply:
Don't know. We were discussing with Nicola and maybe it makes sense to have those generic commands that allow to list and run also containers that are not managed by the framework.